src/Microsoft.Azure.Functions.Extensions.Mcp/Backplane/BackplaneService.cs

```diff
@@ -43,14 +43,16 @@ private async Task InitializeBackplaneProcessing(ChannelReader<McpBackplaneMessa
             }
         }
 
-        private Task HandleMessageAsync(string clientId, JsonRpcMessage message, CancellationToken cancellationToken)
+        private async Task HandleMessageAsync(string clientId, JsonRpcMessage message, CancellationToken cancellationToken)
         {
-            if (!sessionManager.TryGetSession(clientId, out IMcpClientSession? session))
+            var result = await sessionManager.TryGetSessionAsync(clientId);
+
+            if (!result.Succeeded)
             {
                 throw new InvalidOperationException($"Invalid client id. The session for client '{clientId}' does not exist in instance '{_instanceId}'.");
```

Comment:
Do we know for sure the issue is the an invalid client ID? Could something else be happening here?